### PR TITLE
profil page

### PR DESCRIPTION
**Pull request by ramazankani**

### CODE REVIEW

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java [hunk 1]
```diff
@@ -1,15 +1,59 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
+
+        String authStr = getSharedPref(this).getString(Constants.AUTH_STR, Constants.NO_AUTH_STR);
+        APIUtils.serverAPI().getMe(authStr).enqueue(new Callback<UserResponse>() {
+            @Override
+            public void onResponse(Call<UserResponse> call, Response<UserResponse> response) {
+                if (response.isSuccessful()) {
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java]: You can get TextView s out of the if-else statements

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java [hunk 2]
```diff
@@ -1,15 +1,59 @@
...
+                    UserResponse body = response.body();
+                    ((TextView) findViewById(R.id.user_profile_name)).setText(body.getFirstname()+" "+body.getLastname());
+                    ((TextView) findViewById(R.id.user_profile_email)).setText(body.getEmail());
+                } else {
+                    ((TextView) findViewById(R.id.user_profile_name)).setText("name");
+                    ((TextView) findViewById(R.id.user_profile_email)).setText("email");
+                }
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java]: 1. You can get TextView getting out of the if-else statements.
2. We need to show a Toast if the response is not successful.

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java [hunk 3]
```diff
@@ -1,15 +1,59 @@
...
+            }
+            @Override
+            public void onFailure(Call<UserResponse> call, Throwable t) {
+                ((TextView) findViewById(R.id.user_profile_name)).setText("name");
+                ((TextView) findViewById(R.id.user_profile_email)).setText("email");
+            }
+        });
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java]: Need connection failure toast here, as well.

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java [hunk 4]
```diff
@@ -1,15 +1,59 @@
...
+
+        TextView logout = findViewById(R.id.plogout);
+        logout.setOnClickListener((View btnview)-> {
+            // remove token from sharedpref
+            Utils.getSharedPrefEditor(this).remove(Constants.AUTH_STR).apply();
+            // go to authentication activity
+            Intent intent = new Intent(this, AuthActivity.class);
+            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK|Intent.FLAG_ACTIVITY_NEW_TASK);
+            this.startActivity(intent);
+        });
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/profil/ProfileActivity.java]: 1. Can you create a function ```public static void logout(Context context)``` in ```Utils.java``` that does logout handling?
2. We also use these same lines of code in ```HomeActivity NavigationDrawer```. Can you call ```logout``` function after refactoring these lines to ```logout``` function.

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 5]
```diff
@@ -1,17 +1,144 @@
...
+
+            <TextView
+                android:id="@+id/user_profile_email"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/user_profile_name"
+                android:layout_centerHorizontal="true"
+                android:layout_marginTop="12dp"
+                android:text="ramazankaniturkmen@gmail.com"
+                android:textColor="#fff"
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: We shouldn't have any e-mail as android:text. You should write simply **e-mail** there.

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 6]
```diff
@@ -1,17 +1,144 @@
...
+                android:layout_marginTop="16dp"/>
+
+            <TextView
+                android:id="@+id/user_profile_name"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_centerHorizontal="true"
+                android:layout_marginTop="76dp"
+                android:text="Ramazan Kani Türkmen"
+                android:textColor="#fff"
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: Change android:text to simply **name**.

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 7]
```diff
@@ -1,17 +1,144 @@
...
+
+            <TextView
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_margin="5dp"
+                android:background="#fff"
+                android:clickable="true"
+                android:elevation="4dp"
+                android:padding="20dp"
+                android:text="My Culturel Heritages"/>
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: 1. Can you move all raw texts such as this to strings.xml and give a reference here?
2. Also, culturel should be cultural.

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 8]
```diff
@@ -1,17 +1,144 @@
...
+
+            <TextView
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_margin="5dp"
+                android:background="#fff"
+                android:clickable="true"
+                android:elevation="4dp"
+                android:padding="20dp"
+                android:text="My Favorites" />
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: To strings.xml

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 9]
```diff
@@ -1,17 +1,144 @@
...
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_margin="5dp"
+                android:layout_marginBottom="3dp"
+                android:layout_marginTop="3dp"
+                android:background="#fff"
+                android:clickable="true"
+                android:elevation="4dp"
+                android:padding="20dp"
+                android:text="Settings" />
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: To strings.xml

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 10]
```diff
@@ -1,17 +1,144 @@
...
+            <TextView
+                android:id="@+id/plogout"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_margin="5dp"
+                android:background="#fff"
+                android:clickable="true"
+                android:elevation="4dp"
+                android:padding="20dp"
+                android:text="Logout" />
```
Comment (eozd) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml]: Again, to strings.xml

File: Atlas/android/app/src/main/res/layout/activity_profile.xml [hunk 11]
```diff
@@ -1,17 +1,144 @@
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:id="@+id/home_container"
```
Comment (mmutas) [inline Atlas/android/app/src/main/res/layout/activity_profile.xml:3]: You need to select another id.

**Review by eozd — APPROVED** [review]
I have fetched and tested these changes; they seem OK for now. I am merging this

**Pull request merged**